catroid/src/main/java/org/catrobat/catroid/utils/CrashReporter.java

```diff
@@ -96,29 +88,15 @@ public static boolean logException(Throwable t) {
 		return false;
 	}
 
-	public static boolean checkIfCrashRecoveryAndFinishActivity(final Activity context) {
-		Log.d(TAG, "AFTER_EXCEPTION : checkIfCrashRecoveryAndFinishActivity()");
-		if (isRecoveredFromCrash()) {
-			if (isReportingEnabled()) {
-				sendUnhandledCaughtException();
-			}
-			if (!(context instanceof MainMenuActivity)) {
-				context.finish();
-			} else {
-				preferences.edit().putBoolean(RECOVERED_FROM_CRASH, false).commit();
-				return true;
-			}
-		}
-		return false;
-	}
-
 	public static void sendUnhandledCaughtException() {
-		Log.d(TAG, "AFTER_EXCEPTION : sendCaughtException()");
-		Gson gson = new Gson();
 		String json = preferences.getString(EXCEPTION_FOR_REPORT, "");
-		Throwable exception = gson.fromJson(json, Throwable.class);
-		if (logException(exception)) {
-			preferences.edit().remove(EXCEPTION_FOR_REPORT).commit();
+		if (isReportingEnabled() && !json.isEmpty()) {
+			Log.d(TAG, "AFTER_EXCEPTION : sendCaughtException()");
+			Gson gson = new Gson();
+			Throwable exception = gson.fromJson(json, Throwable.class);
+			if (logException(exception)) {
```

Comment:
We can not remove the return value because it is [utilized in the unit test](https://github.com/nishantt95/Catroid/blob/7a78857c1d88f12c5260f071abb5663c781b130f/catroid/src/androidTest/java/org/catrobat/catroid/test/utils/CrashReporterTest.java#L134-L147). But yes, we are already using `isReportingEnabled()` so we don't need to apply if condition here.